### ISSUE TITLE
TriangleTreeTests#testRandomMultiLineIntersections fails reproducibly

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#56755]
**Build scan**:

https://gradle-enterprise.elastic.co/s/nzxu4rmpyax4k

**Repro line**:

```
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.index.fielddata.TriangleTreeTests.testRandomMultiLineIntersections" -Dtests.seed=335F1970924440A6 -Dtests.security.manager=true -Dtests.locale=fr-BE -Dtests.timezone=Indian/Maldives -Dcompiler.java=14 -Druntime.java=8
```

**Reproduces locally?**:

Yes

**Applicable branches**:

`7.x` (b87aeb09f74cb851d0b27a9b4b1f8ed134278258); I have not investigated other branches.

**Failure excerpt**:

```
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.index.fielddata.TriangleTreeTests.testRandomMultiLineIntersections" -Dtests.seed=335F1970924440A6 -Dtests.security.manager=true -Dtests.locale=fr-BE -Dtests.timezone=Indian/Maldives -Dcompiler.java=14 -Druntime.java=8
  2> java.lang.IllegalArgumentException: invalid longitude NaN; must be between -180.0 and 180.0
        at __randomizedtesting.SeedInfo.seed([335F1970924440A6:6424041013FB5688]:0)
        at org.apache.lucene.geo.GeoUtils.checkLongitude(GeoUtils.java:73)
        at org.apache.lucene.geo.GeoEncodingUtils.encodeLongitude(GeoEncodingUtils.java:91)
        at org.elasticsearch.xpack.spatial.index.fielddata.GeoShapeCoordinateEncoder.encodeX(GeoShapeCoordinateEncoder.java:22)
        at org.elasticsearch.xpack.spatial.index.fielddata.TriangleTreeWriter.writeTo(TriangleTreeWriter.java:38)
        at org.elasticsearch.xpack.spatial.index.fielddata.TriangleTreeTests.triangleTreeReader(TriangleTreeTests.java:431)
        at org.elasticsearch.xpack.spatial.index.fielddata.TriangleTreeTests.testRandomMultiLineIntersections(TriangleTreeTests.java:266)
```

**Comment by DaveCTurner** [comment]
I see three recent failures of this in the build stats, all on `7.x`, so I have muted this test on `7.x` (only) with bec6821fe61.

**Comment by iverase** [comment]
fixed in #58216

**Comment by cbuescher** [comment]
This still seems to fail on 7.8: https://gradle-enterprise.elastic.co/s/2cek73xlv5yze
@iverase  can we can backport the fix you mentioned above to that branch as well or should it already be fixed on 7.8?

**Comment by iverase** [comment]
I am closing this as 7.9 has been released and this issue only happens in 7.8 branch
